code/modules/mob/living/living_say.dm

```diff
@@ -211,6 +211,10 @@ GLOBAL_LIST_INIT(message_modes_stat_limits, list(
 		if(succumbed)
 			succumb()
 		return
+	
+	//Updates the verb variables of this living before sending any messages
+	//This avoids copying the wrong verbs to the radio's virtualspeaker
+	update_verbs(message, message_mods)
```

Comment:
Though it seems a little... excessive to call this every time regardless of if `verb_say` has changed? 

Why can we not just update `verb_say` of the mob when it is necessary to - ie when inserting / removing a tongue, or when the tongue's temp say mod changes, or when certain traits are added or remove?

Reply:
Ah, the previous method was already called every time a message is _composed_, regardless of if `verb_say` had actually changed. Most commonly I believe this would be called from `hear() > compose_message() > say_quote() > say_mod()`.
So given I felt running it whenever you say something is still broadly saner than running it for anyone who hears your message, I didn't think it'd be a significant issue for this pr.

That aside, it _is_ probably a better idea.
I believe for `human.dm` instances it only matters when the tongue gets added/removed or the `temp_say_mod` gets changed by sign language (the only thing which changes it). So looking into that deeper, I think it'd be fine to call it on `Insert()` and `Remove()` in `_tongue.dm` and on the sign language component when it would change the speaker's `verb_X`s. `Insert()` and `Remove()` should also be after the `COMSIG_CARBON_GAIN/LOSE_ORGAN` signals and so update it after sign language's changes.

Now for the caveats, there's the basic/simple animals which used it to update `verb_say` like this:
```byond
/mob/living/simple_animal/say_mod(input, list/message_mods = list())
	if(length(speak_emote))
		verb_say = pick(speak_emote)
	return ..()
```
Which would have the same issues, though significantly less common. But there's also no easy "update it only on changes" cause it's _intended_ to change every say attempt.
So maybe updating it at the end of `say()` if it returns `TRUE` in `basic.dm` and `simple_animal.dm` would be fine?
Like say:
```byond
/mob/living/simple_animal/say(pile of arguments)
	. = ..()
	if (.)
		update_say_verb()
	return .
```

I'll be testing such when I have more time tomorrow.

Reply:
Had some circumstances so it took me a while to get back to you.

On the actual topic, hmn, so that sounds good, and yes virtualspeaker is where the original issue lies, but there's an immediate issue I see with this implementation regarding simple_animal/basic.
Because they use `pick()`, `get_default_say_verb()` will pick a _random_ say verb each time and thus the virtualspeaker verb mismatch issue wouldn't be solved for them as it might return a different verb for the virtualspeaker initialization than for the initial message.
![image](https://github.com/tgstation/tgstation/assets/42909981/4832a035-bfd1-41da-acbe-a8b5bbbbbd00)
Even causing this to be a new issue regarding simple_mob/basic instances speaking into radios/somesuch at all
Besides that I feel it'd be broadly better, there's only a few more `verb_say` reading occurrences we'd have to change, and also tested it and it works fine. I'll have to sit down and think of a good solution regarding the simple_mob/basic issue in a bit though.

On the final bit, do we even _have_ access modifiers? I can't find any in the reference nor have i seen any. Do we have linting for it? Regardless I feel it's probably important we update `code/modules/mob/say_readme.md` to match this change.,

Reply:
One of the `verb_say` reading occurrences is in `/obj/machinery/ai_voicechanger`, which reads its owner AI's `verb_say`, and upon closer inspection seems to do so for storing it before replacing it with a different one.
Under the proposed system, I feel we'd need to gut the current logic and instead have the `ai_voicechanger` instance store only the verbs it's changing into while adding the verb selection logic to the owner AI's `say_mod()` and `get_default_say_verb()`.

Reply:
With irl circumstances almost over, just one-two more days to go and I can get back to this.
I care about it cause under the proposed logic it'd it'd be _more_ broken than before.
Before it would only update `verb_say` after the speak-into-radio part and thus the virtualspeaker created before then would have the wrong `verb_say`, but with the proposed logic creating a virtualspeaker for a basic/simple animal is guaranteed to pick a random say verb regardless- so including when they're heard over radios _at all_. Hell, I think it'd even cause the say verb to randomly be different for different listeners regardless of radio.

Y'know, I feel like this'd all be much less of a hassle if the verb used was included in the message data rather than retrieved from the speaker with every listener, so there is one definitive point where the verb is decided on and then it just carries over. Entirely removing the need to worry about whether the say verb we're accessing is still the right say verb in the first place.
There's probably some catches to this, like talking into a radio directly making you whisper, but I'll have to wait with looking deeper into the viability of such for a few more days.